src/core/experimental/memutils.d

```diff
@@ -0,0 +1,685 @@
+/**
+ * Pure D replacement of the C Standard Library basic memory building blocks of string.h
+ * Source: $(DRUNTIMESRC core/experimental/memutils.d)
+ */
+module core.experimental.memutils;
+
+void memmove(T)(T *dst, const T *src)
```

Comment:
Needs full Ddoc comment.
Also, what's the reason for not reason `ref`?

Reply:
Also, thinking more about this interface `src, dst` would be more D-like, but I guess we can't really change that.